src/Bridge/Doctrine/EventListener/PurgeHttpCacheListener.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace ApiPlatform\Core\Bridge\Doctrine\EventListener;
+
+use ApiPlatform\Core\Api\IriConverterInterface;
+use ApiPlatform\Core\Exception\InvalidArgumentException;
+use ApiPlatform\Core\HttpCache\ResourceManagerInterface;
+use Doctrine\ORM\Event\OnFlushEventArgs;
+use Psr\Container\ContainerInterface;
+
+/**
+ * Stores IRIs of entity to purge from the proxy cache.
+ *
+ * @author Kévin Dunglas <dunglas@gmail.com>
+ *
+ * @internal
+ */
+final class PurgeHttpCacheListener
+{
+    private $container;
+
+    /**
+     * @param ContainerInterface|\Symfony\Component\DependencyInjection\ContainerInterface $container The container is injected because of a circular reference.
+     */
+    public function __construct($container)
+    {
+        $this->container = $container;
+    }
+
+    public function onFlush(OnFlushEventArgs $eventArgs)
+    {
+        $iriConverter = $this->container->get('api_platform.iri_converter');
+        $resourceManager = $this->container->get('api_platform.http_cache.resource_manager');
+        $uow = $eventArgs->getEntityManager()->getUnitOfWork();
+
+        foreach ($uow->getScheduledEntityInsertions() as $entity) {
+            $this->purge($iriConverter, $resourceManager, $entity, false);
```

Comment:
are you sure to be able to retrieves an auto-increment id in the `onFlush` eventListener? I thought, this ID was injected *after* the insert operation